lib/executor.js

```diff
@@ -1,57 +1,103 @@
+  }
+
+  /*
+  Generates the QDM patients for calculation
+  @params [array] patientIds -- the patients to generate
+  @returns [Promise] a promise of executable patients
+
+  */
+  async getPatientsById(patientIds) {
+    // patientIds parameter will either be a single id or multiple ids.
+    // If the former, transform it to an array
+    const patientIdsList = Array.isArray(patientIds) ? patientIds : [patientIds];
+
+    return this.patientSource.findPatients(patientIdsList);
   }
 
   /*
    Executes measures against a given patient source
-   @params [PatientSource] patientSource -- the source for patient records.
+   @params {array} patientIds -- the array of patient ids for which results are requested. The patients all adhere to the QDM data model in MongoDB
 
    @params {array} measureIds -- the array of measures to execute. The ids can be CMS ids with the included sub_id if applicable, the hqmfId with the
    included sub_id if applicable or a json object that contains {measure_id: (CMS or HQMF_ID), sub_id: "" }
 
-   @params {object} handler -- the pateint level results handler.  During calculation patient level results will be given to this object to handle in some fashion
+   @params {object} options -- json object with calculation option such as handler information, etc ....
+
+   @params {object} options - handler information -- the pateint level results handler.  During calculation patient level results will be given to this object to handle in some fashion
    such as saving to a database.  The handler needs to implement the start() finsihed() and handleResult(p_level_result) methods.  start() should basically reset the handler and
    get it ready for calculation, handleResult(presults) will perfrom some task on the handler and finished() allows the handler to know when the paitent
    level calculations are finsihed and it can perform any post processing that needs to be perfromed, such as aggregating results
 
-   @params {object} options -- json object with calculation option ssuch as effective_date , enable_logging ....
    */
-  execute(patientSource, measureIds, handler, options) {
-    handler.start();
-    const measures = this.generate_measures(measureIds, options);
+  async execute(patientIds, measureIds, options = null) {
+    await this.getPatientsById(patientIds);
+    const measures = await this.getMeasuresById(measureIds);
+
+    const executor = this;
+
+    let handler;
+
+    if (!options || !options.handlerType || options.handlerType === 'CypressMongo') {
+      handler = new CypressHandler(this.connectionInfo);
+    }
+
+    handler.start(options);
+    await this.valueSetSource.findValueSetsByMongoidForMeasures(measures, (self) => {
+      executor.executeForEachMeasure(measures, self.valueSetsByMongoid, handler, options);
+    });
+    return handler.finish();
+  }
+
+  executeForEachMeasure(measures, valueSetsByMongoid, handler, options) {
     measures.forEach((mes) => {
```

Comment:
I am just starting to wrap my head around how this repo is structured so I am not ready to add a bunch of comments. However, I believe "mes" should be changed to something more specific like "measure"

Reply:
Originally I was debating between using "measure" or "mes". I think I ended up choosing "mes" because I felt it would be a more readable distinction from multiple "measures". Either is fine, just wanted to make aware of the original thought process.

(Also, hi! I'm not active on the project anymore but I'm still here to answer questions if any come up!)